pkg/controllers/uiplugin/controller.go

```diff
@@ -0,0 +1,280 @@
+package uiplugin
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	osv1alpha1 "github.com/openshift/api/console/v1alpha1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metaerrors "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
+	"github.com/rhobs/observability-operator/pkg/reconciler"
+)
+
+type resourceManager struct {
+	k8sClient      client.Client
+	scheme         *runtime.Scheme
+	logger         logr.Logger
+	controller     controller.Controller
+	pluginConf     UIPluginsConfiguration
+	clusterVersion string
+}
+
+type UIPluginsConfiguration struct {
+	Images             map[string]string
+	ResourcesNamespace string
+}
+
+type Options struct {
+	PluginsConf UIPluginsConfiguration
+}
+
+const (
+	AvailableReason         = "UIPluginAvailable"
+	ReconciledReason        = "UIPluginReconciled"
+	FailedToReconcileReason = "UIPluginFailedToReconciled"
+	ReconciledMessage       = "Plugin reconciled successfully"
+	NoReason                = "None"
+)
+
+// RBAC for managing UIPlugins
+// +kubebuilder:rbac:groups=observability.openshift.io,resources=uiplugins,verbs=get;list;watch;create;update;delete;patch
+// +kubebuilder:rbac:groups=observability.openshift.io,resources=uiplugins/status,verbs=get;update
+
+// RBAC for managing observability ui plugin objects
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles;rolebindings,verbs=list;watch;create;update;delete;patch
+//+kubebuilder:rbac:groups="",resources=serviceaccounts;services;configmaps,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Do we really need create/update/patch/delete permissions?
It can be adjusted as a follow-up and in fact, we probably want to review/tidy up the operator permissions after the PR merges.

Reply:
I created a follow up task https://issues.redhat.com/browse/COO-121